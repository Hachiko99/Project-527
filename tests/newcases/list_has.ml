type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

type bool =
  | True
  | False


let list_has : nat -> list -> bool |>
  { 0 => ( [] => False
         | [0] => True
         | [0; 1] => True
         | [1] => False)
  | 1 => ( [] => False
         | [0; 2] => False
         | [0; 1] => True)} = ?
