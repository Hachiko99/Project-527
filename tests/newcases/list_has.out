(* Generated Tue Apr 19 12:38:28 2016 *)
let list_has : nat -> list -> bool =
  let rec f1 (n1:nat) : list -> bool =
    fun (l1:list) ->
      match n1 with
        | O -> (match l1 with
                  | Nil -> False
                  | Cons (n2, l2) -> (match n2 with
                                        | O -> True
                                        | S (n3) -> False))
        | S (n2) -> (match f1 n2 l1 with
                       | True -> True
                       | False -> False)
  in
    f1
;;
